packages/migration/src/migrations/hearth/20241029171702-add-certificateTemplateId-to-missing-docs.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import { Db, MongoClient } from 'mongodb'
+
+// Define an interface for your DocumentReference
+interface DocumentReference {
+  _id: string
+  extension: Array<{ url: string; valueString?: string }>
+  type?: {
+    coding?: Array<{ system: string; code: string }>
+  }
+}
+
+export const up = async (db: Db, client: MongoClient) => {
```

Comment:
ok, let me try this.

Reply:
Tried, there are little bit of correction in query:
```
[
        {
          $match: {
            'extension.url': {
              $ne: 'http://opencrvs.org/specs/extension/certificateTemplateId'
            }
          }
        },
        {
          $set: {
            certType: {
              $arrayElemAt: [
                {
                  $filter: {
                    input: '$type.coding',
                    as: 'coding',
                    cond: {
                      $regexMatch: {
                        input: '$$coding.system',
                        regex: /certificate-type/
                      }
                    }
                  }
                },
                0
              ]
            }
          }
        },
        {
          $set: {
            certificateTemplateId: {
              $switch: {
                branches: [
                  {
                    case: { $eq: ['$certType.code', 'BIRTH'] },
                    then: 'birth-certificate'
                  },
                  {
                    case: { $eq: ['$certType.code', 'DEATH'] },
                    then: 'death-certificate'
                  },
                  {
                    case: { $eq: ['$certType.code', 'MARRIAGE'] },
                    then: 'marriage-certificate'
                  }
                ],
                default: null
              }
            }
          }
        },
        {
          $match: {
            certificateTemplateId: { $ne: null }
          }
        },
        {
          $set: {
            extension: {
              $concatArrays: [
                '$extension',
                [
                  {
                    url: 'http://opencrvs.org/specs/extension/certificateTemplateId',
                    valueString: '$certificateTemplateId'
                  }
                ]
              ]
            }
          }
        },
        {
          $unset: ['certType', 'certificateTemplateId']
        },
        {
          $merge: {
            into: 'DocumentReference',
            whenMatched: 'replace'
          }
        }
      ]
```

I had to unset the certType and certificateTemplateId so that they don't get added in the document as new propertis.

Reply:
Great, I think that works